### PR TITLE
Add vsphere conformance micro upgrade jobs for 4.14

### PR DESCRIPTION
**Pull request by DennisPeriquet**
[TRT-1160](https://issues.redhat.com//browse/TRT-1160)

This is just for 4.14.

There are two items (in a commit) we may want to revisit:
* one to add the job for 4.15 but it's not passing rehearse yet.  I can wait on that until we get closer to (or after moving to 4.15).
* one to add the `gather-core-dump` by modifying the workflow (I held off on this to avoid changing 86 jobs as a result).

### CODE REVIEW

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.14.yaml [hunk 1]
```diff
@@ -422,6 +422,16 @@ tests:
       enable:
       - observers-resource-watch
     workflow: openshift-e2e-aws-sdn-crun
+- as: e2e-vsphere-ovn
```
Comment (neisw) [inline ci-operator/config/openshift/release/openshift-release-master__ci-4.14.yaml:425]: Title reference adding micro job, should this be an upgrade?  Do we need the same in 4.15?

**Comment by neisw** [comment]
https://github.com/openshift/release/pull/42521 created the minor upgrades, in case you haven't already seen it.

File: ci-operator/config/openshift/release/openshift-release-master__ci-4.14.yaml (hunk 1 above)
Comment (DennisPeriquet) [reply ci-operator/config/openshift/release/openshift-release-master__ci-4.14.yaml:425]: thanks.  Will make it an upgrade job and also add one for 4.15.

**Comment by DennisPeriquet** [comment]
pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-release-4.14-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
/pj-rehearse pull-ci-openshift-cloud-provider-vsphere-release-4.14-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.14-e2e-vsphere-ovn-upgrade

**Comment by DennisPeriquet** [comment]
I keep getting this symptom on the rehearse jobs, [example](https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_release/42530/rehearse-42530-periodic-ci-openshift-release-master-ci-4.14-e2e-vsphere-ovn-upgrade/1699400066688618496):

```
INFO[2023-09-06T13:13:06Z] pod pending for more than 30m0s: containers have not started in 30m0.000783201s: test: 
* Container test is not ready with reason ImagePullBackOff and message Back-off pulling image "image-registry.openshift-image-registry.svc:5000/ci-op-j3qhdc5f/stable:upi-installer"
```

**Comment by DennisPeriquet** [comment]
/pj-rehearse periodic-ci-openshift-release-master-ci-4.15-e2e-vsphere-ovn-upgrade

**Review by vrutkovs — APPROVED** [review]
/approve
/lgtm

**Pull request closed without merging**